app/models/user.rb

```diff
@@ -1,25 +1,63 @@
 class User < ApplicationRecord
   include Cul::Omniauth::Users
-  ROLES = %i[admin superadmin]
+
+  has_many :permissions, dependent: :destroy
+
   before_validation :add_email
 
   def password
     Devise.friendly_token[0,20]
   end
 
   def password=(*val)
-
   end
 
   def admin?
-    self.role == "admin"
+    can? :manage, :all
+  end
+
+  def editor?
+    !editable_libraries.count.zero?
   end
 
-  def superadmin?
-    self.role == "superadmin"
+  def editable_libraries
+    self.permissions
+      .where(action: 'edit', subject_class: Library.to_s)
```

Comment:
Using constants is always better, and this saves me having to create a constant. It also assures that there isn't a typo in the class name.